core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -223,6 +224,63 @@ class TransactionStateManager(brokerId: Int,
       throw new IllegalStateException(s"Unexpected empty transaction metadata returned while putting $txnMetadata")))
   }
 
+  def listTransactionStates(
+    filterProducerIds: Set[Long],
+    filterStateNames: Set[String]
+  ): ListTransactionsResponseData = {
+    inReadLock(stateLock) {
+      val response = new ListTransactionsResponseData()
+      if (loadingPartitions.nonEmpty) {
+        response.setErrorCode(Errors.COORDINATOR_LOAD_IN_PROGRESS.code)
+      } else {
+        val filterStates = mutable.Set.empty[TransactionState]
+        val unknownStates = new java.util.ArrayList[String]
+        filterStateNames.foreach { stateName =>
+          TransactionState.fromName(stateName) match {
+            case Some(state) => filterStates += state
+            case None => unknownStates.add(stateName)
```

Comment:
It response#unknownStates is NOT nullable, we can simplify it to `case None => response.unknownStateFilters().add(stateName)`

Reply:
Yeah, I agree. It is simpler. I guess I don't have a strong reason to make this field nullable.